runner/runner/listener.py

```diff
@@ -59,6 +62,107 @@ def _handle_delivery(channel, deliver, properties, body):
             case _:
                 logger.error("Unrecognized message type: %s", msg_type)
 
-        channel.basic_ack(deliver.delivery_tag)
+        channel.basic_ack(method.delivery_tag)
     except Exception as exc:
         logger.error("Error handling received message: %s", exc)
+
+
+def _get_inspect() -> Inspect:
+    """Return Celery Inspect object after connection to Celery Workers is established"""
+    inspect = Inspect(app=app)
+    while (workers := inspect.ping()) is None:
+        logger.info("Waiting for connection to Celery workers to be established.")
+        sleep(1)
+
+    logger.info(f"Number of available workers: {len(workers.keys())}")
+    return inspect
+
+
+def _create_inspect() -> None:
+    """Create global Celery Inspect object to interact with Celery Worker API"""
+    global celery_inspect
+    celery_inspect = _get_inspect()
+
+
+@cached(cache=TTL_CACHE)
```

Comment:
Ok, makes sense. I'll need to slightly alter the code that checks if the worker is available so that it is just checking the worker's number of assigned tasks to that `WORKER_CONCURRENCY`, rather than a dictionary of the worker's concurrency levels. I'll also probably keep that `_get_worker_concurrency()` method just for the sake of unit testing.